Tasks/DownloadBuildArtifact/task.json

```diff
@@ -0,0 +1,59 @@
+{
+  "id": "a433f589-fce1-4460-9ee6-44a624aeb1fb",
+  "name": "DownloadBuildArtifact",
+  "friendlyName": "Download Build Artifact",
+  "description": "Download a build artifact.",
+  "helpMarkDown": "",
+  "category": "Utility",
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 0,
+    "Minor": 1,
+    "Patch": 48
+  },
+  "demands": [],
+  "inputs": [
+    {
+      "name": "buildId",
+      "type": "pickList",
+      "label": "Build",
+      "required": false,
+      "helpMarkDown": "The build from which to download the artifact",
+      "defaultValue": "$(Build.BuildId)",
+      "options": {
+        "$(Build.BuildId)": "The current build"
+      }
+    },
+    {
+      "name": "artifactName",
+      "type": "string",
+      "label": "Artifact name",
+      "defaultValue": "drop",
+      "required": true,
+      "helpMarkDown": "The name of the artifact to download"
+    },
+    {
+      "name": "downloadPath",
+      "type": "string",
+      "label": "Destination directory",
+      "defaultValue": "$(System.ArtifactsDirectory)",
```

Comment:
That seems like a bug.  It is a System.* variable.  It also seems problematic to hope all extensions set something like that.  Shouldn't the agent just set it since it's a system level thing?  The agent should define the a dir then RM extension should respect and use it.